common/account_utils/src/lib.rs

```diff
@@ -121,9 +123,26 @@ pub fn read_mnemonic_from_user(use_stdin: bool) -> Result<String, String> {
             .read_line(&mut input)
             .map_err(|e| format!("Error reading from tty: {}", e))?;
     }
+    trim_newline(&mut input);
     Ok(input)
 }
 
+fn trim_newline(s: &mut String) {
+    if s.ends_with('\n') {
+        s.pop();
+        if s.ends_with('\r') {
+            s.pop();
+        }
+    }
+}
+
+/// Takes a string password and checks that it meets minimum requirements.
+///
+/// The current minimum password requirement is a 12 character length character length.
+pub fn is_password_sufficiently_complex(password: &[u8]) -> bool {
```

Comment:
It's not very important, but if this returned `Result<(), String>` you could avoid repeating `"Please use at least {} characters for your password."` in downstream functions by passing that message via the `Result::Err(String)`.

Reply:
That's smart, easier to update error messages if the password requirements change too.